File: Python/0026-e-remove-duplicates-from-sorted-array.py
# Time: O(n)
# Space: O(1)
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0: 
            return 0 
        index = 0 
        for i in nums[1:]: 
            if i != nums[index]: 
                index += 1 
                nums[index] = i
        return index + 1

File: leetcode/C++/0026-e-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty())
            return 0;
        int cur = 0;
        for (int i = 0; i < nums.size(); i++) {
            nums[cur++] = nums[i];
            if (cur >= 2 && nums[cur-1] == nums[cur-2])
                cur--;
        }
        return cur;
    }
};

class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        return distance(nums.begin(), unique(nums.begin(), nums.end()));
    }
};